framework/test_selectors/SelectorFactory.cpp

```diff
@@ -188,17 +189,51 @@ struct test *TestrunSelector::testid_to_test(const char *id, bool silent)
     return nullptr;
 }
 
+namespace {
+static constexpr auto arch_spr = "spr";
+static constexpr auto arch_icx = "icx";
+static constexpr auto arch_cpx = "cpx";
+static constexpr auto arch_skx = "skx";
+
+// source https://en.wikichip.org/wiki/intel/cpuid
+const char* get_test_list_name_from_cpu_info(const struct cpu_info& info) {
+    if (info.family != 0x6) {
+        // so far only 0x6 family processors have dedicated test lists
+        return nullptr;
+    }
+
+    if (info.model == 0x8F) {
+        return arch_spr;
+    } else if (info.model == 0x6C || info.model == 0x6A) {
+        return arch_icx;
+    } else if (info.model == 0x55 && info.stepping >= 7) { // TODO stepping: is this true?
+        return arch_cpx;
+    } else if (info.model == 0x55 && info.stepping < 7) { // TODO
+        return arch_skx;
+    } else {
+        return nullptr;
+    }
+}
+}
+
 void generate_test_list(std::vector<struct test *> &test_list, std::span<struct test> test_set,
                         int min_quality)
 {
     if (SandstoneConfig::RestrictedCommandLine || test_list.empty()) {
         if (!SandstoneConfig::RestrictedCommandLine && sApp->fatal_skips)
             fprintf(stderr, "# WARNING: --fatal-skips used with full test suite. This will probably fail.\n"
                             "# You may want to specify a controlled list of tests to run.\n");
-        /* generate test list based on quality levels only */
-        for (struct test &test : test_set) {
-            if (test.quality_level >= min_quality)
-                add_test(test_list, &test);
+        auto builtin_test_list = get_test_list(get_test_list_name_from_cpu_info(sApp->shmem->cpu_info[0]), true); // set fallback to default to true
+        if (builtin_test_list) {
+            for (auto* test : *builtin_test_list) {
+                if (test->quality_level >= min_quality)
```

Comment:
seems that "sometimes" nullptr might be passed in the list. Add appropriate check to avoid SEGFAULTS.